mitiq/interface/mitiq_pennylane/conversions.py

```diff
@@ -0,0 +1,87 @@
+# Copyright (C) 2021 Unitary Fund
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+"""Functions to convert between Mitiq's internal circuit representation and
+Pennylane's circuit representation.
+"""
+
+from cirq import Circuit, Moment
+
+from mitiq.interface.mitiq_qiskit import from_qasm as cirq_from_qasm, to_qasm
+from mitiq.utils import _pop_measurements
+
+from pennylane.wires import Wires
+from pennylane.tape import QuantumTape
+from pennylane.operation import Expectation
+
+from pennylane import from_qasm as pennylane_from_qasm
+
+
+class UnsupportedQuantumTapeError(Exception):
+    pass
+
+
+def from_pennylane(tape: QuantumTape) -> Circuit:
+    """Returns a Mitiq circuit equivalent to the input QuantumTape.
+
+    Args:
+        tape: Pennylane QuantumTape to convert to a Mitiq circuit.
+
+    Returns:
+        Mitiq circuit representation equivalent to the input QuantumTape.
+    """
+    try:
+        wires = sorted(tape.wires)
+    except TypeError:
+        raise UnsupportedQuantumTapeError(
+            f"The wires of the tape must be sortable, but could not sort "
+            f"{tape.wires}."
+        )
+
+    for i in range(len(wires)):
+        if wires[i] != i:
+            raise UnsupportedQuantumTapeError(
+                "The wire labels of the tape must contiguously pack 0 "
+                "to n-1, for n wires."
+            )
+
+    if not all(m.return_type is Expectation for m in tape.measurements):
+        raise UnsupportedQuantumTapeError(
+            "Only expectation value measurements are supported."
+        )
+    output = cirq_from_qasm(tape.to_openqasm(rotations=True, wires=wires))
```

Comment:
```suggestion
    tape = tape.expand(stop_at=lambda obj: obj.name in SUPPORTED)
    
    output = cirq_from_qasm(tape.to_openqasm(rotations=True, wires=wires))
```
This expands the tape to only supported operations.

Reply:
Cool!